aiida_wannier90/calculations.py

```diff
@@ -187,14 +195,48 @@ def _SEEDNAME(self):
         Return the default seedname, unless a custom one has been set in the
         calculation settings
         """
-        return self.inputs.metadata.options.seedname
+        input_filename = self.inputs.metadata.options.input_filename
+
+        if input_filename.endswith(self._REQUIRED_INPUT_SUFFIX):
+            return input_filename[:-len(self._REQUIRED_INPUT_SUFFIX)]
+        # This would be an invalid input filename. However, I do not raise here since this is a property.
+        # I just return the full input_filename, but there is some validation in prepare_for_submission.
+        return input_filename
 
     def prepare_for_submission(self, folder):  #pylint: disable=too-many-locals, too-many-statements # noqa:  disable=MC0001
         """
         Routine which creates the input file of Wannier90
         :param folder: a aiida.common.folders.Folder subclass where
             the plugin should put all its files.
         """
+        # Let's check that the user-specified input filename ends with .win
+        if not self.inputs.metadata.options.input_filename.endswith(
+            self._REQUIRED_INPUT_SUFFIX
+        ):
+            raise exc.InputValidationError(
+                "The input filename for Wannier90 (specified in the metadata.options.input_filename) "
+                "must end with .win, you specified instead '{}'".format(
+                    self.inputs.metadata.options.input_filename
+                )
+            )
+
+        # The output filename is defined by Wannier90 based on the seedname.
+        # In AiiDA, the output_filename needs to be specified as a metadata.option to allow for
+        # `verdi calcjob outputcat` to work correctly. Here we check that, if the users manually changed
+        # the input_filename, they also changed the output_filename accordingly
+        expected_output_filename = self._SEEDNAME + ".wout"
+        if self.inputs.metadata.options.output_filename != expected_output_filename:
+            raise exc.InputValidationError(
```

Comment:
It is a bit unfortunate that the user will have to modify both input and output filename to change the seedname. For this PR I would leave it as is, but I wonder if we could implement the following logic somehow:
- if neither is specified explicitly, use the defaults
- if only one of the two is specified explicitly, set the other one to use the same seedname
- if both are set, just check that they are consistent

or even have still have a separate ``seedname`` input. The way I'm thinking of implementing this is to have a ``validator`` that would modify the inputs on the fly, to ensure that the ``inputcat`` and ``outputcat`` functionality still works.

@sphuber do you see any obvious issues with such an approach?

In any case, this can be a separate issue, as that could be implemented in a backwards-compatible way.

Reply:
Yes, I agree this is suboptimal. This might be the case also for other codes where the output file name is not arbitrary but depends on the input (it isn't the case in QE were the output is on stdout and we decide on which file it should go, and can have a different prefix than the input file).

I am not sure it is very easy to do what you suggest (but maybe I'm wrong and @sphuber can correct me). But I see your point and we can discuss if this can be done, and then implement it in a backward-compatible way.